Lib/test/test_fstring.py

```diff
@@ -300,6 +301,9 @@ def test_missing_expression(self):
                              "f'{:x'",
                              ])
 
+        self.assertAllRaise(SyntaxError, 'invalid character in identifier',
```

Comment:
Can you add a comment here that says that eval("\xa0") raises SyntaxError with the "invalid character in identifier" text? I think that it's important to know that what you're doing here is trying to match the Python parser. (I realize that, with your change, you're "matching" by actually using the Python parser.)

Or, maybe also add 'eval("\xa0")' to the array to make it obvious that we want the same error raised in both cases?